rest-service/manager_rest/storage/resource_models.py

```diff
@@ -590,9 +590,22 @@ def _is_workflow_available(self, workflow):
             return True
         if not rules.get('available', True):
             return False
+        ni_active_rule = rules.get('node_instances_active')
+        if ni_active_rule:
+            return self._node_instances_active_states_match(ni_active_rule)
         # TODO add other availability rules checking here
```

Comment:
We probably shouldn't have a TODO in here

Reply:
why not?  there will be a few more, which I'm going to implement soon.  I'm keeping that as a reminder/placeholder

Reply:
As long as it's gone by release.

I note though that as it stands (without the mapping based change I suggested above, or similar), if there is a ni_active_rule and it states that the workflow is available then the workflow will be available even if any following rules want to return False.